beacon_node/beacon_chain/src/kzg_utils.rs

```diff
@@ -76,3 +135,299 @@ pub fn verify_kzg_proof<E: EthSpec>(
 ) -> Result<bool, KzgError> {
     kzg.verify_kzg_proof(kzg_commitment, &z.0.into(), &y.0.into(), kzg_proof)
 }
+
+/// Build data column sidecars from a signed beacon block and its blobs.
+pub fn build_data_column_sidecars<E: EthSpec>(
+    blobs: &BlobsList<E>,
+    block: &SignedBeaconBlock<E>,
+    kzg: &Kzg,
+    spec: &ChainSpec,
+) -> Result<DataColumnSidecarList<E>, DataColumnSidecarError> {
+    let number_of_columns = spec.number_of_columns;
+    if blobs.is_empty() {
+        return Ok(vec![]);
+    }
+    let kzg_commitments = block
+        .message()
+        .body()
+        .blob_kzg_commitments()
+        .map_err(|_err| DataColumnSidecarError::PreDeneb)?;
+    let kzg_commitments_inclusion_proof = block.message().body().kzg_commitments_merkle_proof()?;
+    let signed_block_header = block.signed_block_header();
+
+    let mut columns = vec![Vec::with_capacity(E::max_blobs_per_block()); number_of_columns];
+    let mut column_kzg_proofs =
+        vec![Vec::with_capacity(E::max_blobs_per_block()); number_of_columns];
+
+    // NOTE: assumes blob sidecars are ordered by index
+    let blob_cells_and_proofs_vec = blobs
+        .into_par_iter()
+        .map(|blob| {
+            let blob = blob
+                .as_ref()
+                .try_into()
+                .expect("blob should have a guaranteed size due to FixedVector");
+            kzg.compute_cells_and_proofs(blob)
+        })
+        .collect::<Result<Vec<_>, KzgError>>()?;
+
+    for (blob_cells, blob_cell_proofs) in blob_cells_and_proofs_vec {
```

Comment:
From here to the end of the method looks like the same logic exists in `reconstruct_data_columns`, maybe we can make a common method

Reply:
Refactored in https://github.com/sigp/lighthouse/pull/6212/commits/9b69f1db20e36a8cad27f0bee8583cb84305f82e. Thanks!